V0.3.9

SaarJunius opened the pull request:
Milestone 0.3.9

Still TODO (@SanderDevisscher @soriadelva)
- [x] Blacklist exotenportaal for certain species (https://github.com/inbo/reporting-rshiny-grofwildjacht/issues/598)
- [x] Pas title/description aan die hier zijn toegevoegd https://github.com/inbo/reporting-rshiny-grofwildjacht/commit/4c2a7c682313f6d655acfbbb0122b6f0ec7bfba9#diff-8f663c7eba66d041276abdd6db87206310de529ccb2e9200057b58b79ccd1747 in de uiText.csv
- [x] https://github.com/inbo/reporting-rshiny-grofwildjacht/issues/596


Additional pictures should be added here https://github.com/inbo/reporting-rshiny-grofwildjacht/tree/v0.3.9/reporting-grofwild/inst/ui/www
- [x] Voeg afbeelding toe voor de "Links" category header ("category-links-header.png")
- [x] Voeg afbeelding toe voor de "Interne links" subcategory ("category-links-internelinks.png")
- [x] Voeg afbeelding toe voor "Vlaams biodiversiteitsportaal" ("category-links-biodiversiteitsportaal.png") 
- [x] Voeg afbeelding toe voor "Exotenportaal" ("category-links-exotenportaal.png")

SanderDevisscher committed:
add gbifkey to species-info.csv

#557

SaarJunius committed:
fix #467 #574 - type was evaluated too early

SaarJunius committed:
feat #528 - Improve search bar

SaarJunius committed:
fix #534 - Searchbox should not overlap items in header

SaarJunius committed:
fix #537 - Improve layout of draagvlak graph

SaarJunius committed:
feat #514 - Restrict referentieperiode options (exclude selected year)

SaarJunius committed:
fix #537 - add dependencies to namespace

SaarJunius committed:
Feat #579 - Maintain sources when switching species

SaarJunius committed:
feat #521 - Split filter for Landkaart > afschot

SaarJunius committed:
feat #536 - Improve navigation part 1

SaarJunius committed:
feat #557 - add logic for gbifkey to url

SaarJunius committed:
feat #536 - Improve navigation part 2

SaarJunius committed:
fix #534 - general layout improvements

SaarJunius committed:
feat #517 - Rework graph "Inschatting kosten"

SaarJunius committed:
feat #573 - Adaptations to "Toekomstig verspreidingsmodel"

SaarJunius committed:
bug #537 - change values of xlabels in surveys (-2,2)

SaarJunius committed:
Merge branch
'557-make-species-page-linkable-by-vbpexotenportaal-inward-linking' into
v0.3.9

SaarJunius committed:
feat #557 - Adapt other 0.3.9 features to url linking

SaarJunius committed:
feat #536 - Vertaal naar Nederlands

SaarJunius committed:
bug #585 - use verbatimCentroidLongitude/Latitude instead of x/y

SaarJunius committed:
fix #587 - Evolutie per leeftijdscat (wbe) only displays common category

SaarJunius committed:
feat #580 - Group Geografische spreiding plots together

SaarJunius committed:
feat #536 - Include scroller in case dropdowns in navbar too long

SaarJunius committed:
bugfix #589 - Handle overflow for tables

SaarJunius committed:
feat #591 - Exclude all 0-cols for tableSchadeCode

SaarJunius committed:
feat #590 - Always start with all types of damage in schade plots

SaarJunius committed:
fix #588 - Re-size countYearSchade graphs after window resizing

SaarJunius committed:
feat #517 - Extra changes to createRawData for schadeBedragOmschrijving

SaarJunius committed:
feat #530 - Reewild app on wbe

SaarJunius committed:
feat #592 - Add hover over jargon terms

SaarJunius committed:
feat #530 - Rename ree categories

SaarJunius committed:
feat #517 - Roxygenize to rm export(modifyQueryString) on correct branch

SaarJunius committed:
feat #517 - Order schadebedrag categories

SaarJunius committed:
feat #517 - Adapt in-plot title

SaarJunius committed:
feat #517 - Adapt in-plot title

SaarJunius committed:
bug #585 - different CRS for verbatimCentroidLat/Long

SaarJunius committed:
feat #530 - Add title and description of Reewild app to uiText

SaarJunius committed:
fix #370 - Labels overlap when lots of bars

SaarJunius committed:
fix #418 - add statsInfo to mapFlanders

SaarJunius committed:
feat #462 - Add percentage option to embryo graph

SaarJunius committed:
feat #593 - Rename filters

SaarJunius committed:
feat #558 - New "links" category

SaarJunius committed:
feat #593 - Add new filters

SaarJunius committed:
feat #593 - Replace graphs to correct subcategory

SaarJunius committed:
feat #593 - Naming of new subcategories

SaarJunius committed:
Administration v0.3.9

SaarJunius committed:
small fixes wbe

SaarJunius committed:
Merge remote-tracking branch 'origin/uat' into v0.3.9

SaarJunius committed:
Resolve merge issues

SaarJunius committed:
feat #598 - Add wild zwijn/edelhert to blacklist of exotenportaal

SanderDevisscher committed:
fix #598

SanderDevisscher committed:
Enhance biodiversity portal description in uiText.csv

Updated the description for the biodiversity portal to provide more detailed information about its purpose and features.

SanderDevisscher committed:
Update description for exotenportaal entry

#599

SanderDevisscher commented:
Deployment van [db9c567](https://github.com/inbo/reporting-rshiny-grofwildjacht/pull/599/commits/db9c5674affe47d17f6c082d07f316283572c7f1) is ongoing, see: https://github.com/inbo/reporting-rshiny-grofwildjacht/actions/runs/18040332816

SanderDevisscher reviewed (dismissed):
## These issues are breaking: 
They should be fixed prior the merger
- [x] "Toekomstige verspreiding" is shown for species that have none (yet) like Canada goose. This might be fixed by merging and uploading the modified files for "Toekomstige verspreiding" which are currently in PR => I'll check. I get the following error when running `createSpreadData()` after installing this version (v0.3.9) of `reportingGrofwild`: 
```
Error in [.data.frame(x, i, j, drop = drop) : 
  undefined columns selected
Called from: [.data.frame(x, i, j, drop = drop)
```
- [x] The text below [Jaarlijkse kosten voor schadegevallen veroorzaakt door ...](https://faunabeheer-uat.inbo.be/app_proxy/5bd288a1-a1c5-4eb6-bbb3-b2d676371060/?specie=Wild%20zwijn&category=schade&subcategory=schade-kosten&plot=barCostUI&gbifkey=7705930#) should be improved. Currently it is `Kost ingeschat door de melder via schadegeval van e-loket ANB. Er is momenteel enkel een kosteninschatting mogelijk bij schadegevallen over landbouw.` @soriadelva can you help me formulate a better description ? The improved text should be added here: https://github.com/inbo/reporting-rshiny-grofwildjacht/blob/db9c5674affe47d17f6c082d07f316283572c7f1/reporting-grofwild/inst/extdata/uiText.csv#L4
- [x] In case of [Toekomstig verspreidingsgebied ...](https://faunabeheer-uat.inbo.be/app_proxy/5bd288a1-a1c5-4eb6-bbb3-b2d676371060/?specie=Wild%20zwijn&category=verspreiding&subcategory=verspreiding-toekomstig&plot=mapSpreadUI&gbifkey=7705930#) the colors for `0%` & `0-20%` should be switched ea 0% should become "white" & 0-20% "light yellow". 
- [x] Somethings are off at [Afschot aanvraag reewild](https://wbe-uat.inbo.be/app_proxy/86a3cf8b-aa6c-4950-9131-3675536acfb4/#).  
    - [x] The bovengrens is lower than the ondergrens in some cases this should never be the case due to the percentages assigned to the age/sex classes.  
    - [x] I've made a mistake in the issue giving the years. Currently it's is `huidig jaar - 1`, `huidig jaar - 2`, `huidig jaar - 3` to determine the request for the next 3y. This should be `huidig jaar - 3`, `huidig jaar - 2`, `huidig jaar - 1` to determine `huidig jaar` (2025) up to `huidig jaar + 2` (2027) when it's **before 15/10/huidig jaar(2025)** & `huidig jaar - 2`, `huidig jaar - 1`, `huidig jaar` to determine `huidig jaar +1` (2026) up to `huidig jaar + 3` (2028) when it's **after 15/10/huidig jaar(2025)**
## These issues are nice to have: 
They should only be fixed if they have a low time investment or they are mostly fixed in function of a breaking issue. If they are not fixed in this PR a new issue should be created. 
- [x] In case of [Jaarlijkse kosten voor schadegevallen veroorzaakt door ...](https://faunabeheer-uat.inbo.be/app_proxy/5bd288a1-a1c5-4eb6-bbb3-b2d676371060/?specie=Wild%20zwijn&category=schade&subcategory=schade-kosten&plot=barCostUI&gbifkey=7705930#) the colors should be inverted `minder dan 300 euro` should become dark blue & `meer dan 3000 euro` should become orange. Also `het schadebedrag is niet bekend` should become grey this more inline with the other graphs where unknown is typically grey  
- [x] Issue #537 is not fixed in case of the Aantrekkingskracht graph. When deselecting a year it is not removed from the graph.

SaarJunius committed:
feat - invert colors of barCost categories

SaarJunius committed:
Merge remote-tracking branch 'origin/v0.3.9' into v0.3.9

SaarJunius committed:
bug #537 - also for aantrekkingskracht graph

SaarJunius committed:
feat #573 - sort names before assigning legend colors

SanderDevisscher commented:
My review is finished. @SaarJunius do you give me a headsup when the breaking issues are fixed ?

SanderDevisscher commented:
@soriadelva can you take a look at the icons and text fragments needed ?

SaarJunius committed:
feat #530 - improvements reewild app

SaarJunius commented:
> My review is finished. @SaarJunius do you give me a headsup when the breaking issues are fixed ?

The changes I can do are implemented and pushed. 
For the createSpreadData error, this is probably also because that pr in the backoffice is not merged yet (we are now expecting a "wildsrt" column that is not yet available in the data). We can check this again after the backoffice is up-to-date.

SanderDevisscher commented:
I see, I've added the wildsrt column to the 2025 files where it was missing. But now I get this error:
```
Error in rbind.data.frame(...) : 
  numbers of columns of arguments do not match
Called from: rbind.data.frame(...)
```

I think the issue happens in these lines:
https://github.com/inbo/reporting-rshiny-grofwildjacht/blob/36251e0305bd2b41354d29fe564d70c3fae8956c/reporting-grofwild/R/data_create.R#L658C7-L662C8

SaarJunius commented:
> I see, I've added the wildsrt column to the 2025 files where it was missing. But now I get this error:
> 
> ```
> Error in rbind.data.frame(...) : 
>   numbers of columns of arguments do not match
> Called from: rbind.data.frame(...)
> ```
> 
> I think the issue happens in these lines: https://github.com/inbo/reporting-rshiny-grofwildjacht/blob/36251e0305bd2b41354d29fe564d70c3fae8956c/reporting-grofwild/R/data_create.R#L658C7-L662C8

Hmm indeed, because the "wildsrt" column is not available in voerenShape, the rbind throws an error. I guess spatialData$communes does not have a "wildsrt" column? (I don't know where you are running this from so I don't see the input data, my apologies).

If not, I suppose we need to think about the path forward: We can manually create a "wildsrt" column in voerenShape, but then we need to know how to fill it:

1. All empty: but I think then voeren will be filtered out anywhere because we filter on species
2. Do a cross-join of the Voeren geomteries with all possible species so that Voeren will be included for all of them
3. Is there a subset of species for which Voeren is relevant, and for others not? If so, we should only cross-join the Voeren geometries with the relevant species

SanderDevisscher commented:
I would go for option 2

SaarJunius committed:
feat #573 - Fix error createSpreadData for Voeren

SaarJunius commented:
> I would go for option 2

I've implemented it such that we duplicate "Voeren" info for every specie that is in the modelShape wildsrt column. It should work now.

SanderDevisscher commented:
Uploading of the data goes as planned now.

SanderDevisscher commented:
@SaarJunius Toekomstige verspreiding is still available for all species while it should currently be only for "Wild zwijn"

soriadelva committed:
Add files via upload

soriadelva committed:
Add category-links-header.png

SaarJunius committed:
feat #573 - Only allow toekomstige verspr for species with data

SaarJunius committed:
Merge branch 'v0.3.9' of https://github.com/inbo/reporting-rshiny-grofwildjacht into v0.3.9

SaarJunius commented:
> @SaarJunius Toekomstige verspreiding is still available for all species while it should currently be only for "Wild zwijn"

Resolved with https://github.com/inbo/reporting-rshiny-grofwildjacht/pull/599/commits/dafcea8ebd4cb144abfaccb6c778fe66c31f9eb0

SaarJunius committed:
feat #558 - correct text on card and description for links

soriadelva committed:
Add category-links png's

soriadelva committed:
Add category-icons.png

soriadelva commented:
I've added the necessary png's to the folder. For the subcategories (the icons with graphs) I've currently added a png called category-icons.png. At the moment, my proposal is to use this one everywhere. We may alter this in the future and adapt it for the different subcategories.

SaarJunius committed:
General icon for subcategory cards

SaarJunius commented:
> I've added the necessary png's to the folder. For the subcategories (the icons with graphs) I've currently added a png called category-icons.png. At the moment, my proposal is to use this one everywhere. We may alter this in the future and adapt it for the different subcategories.

The general "category-icons.png" has been added for all subcategory cards in https://github.com/inbo/reporting-rshiny-grofwildjacht/pull/599/commits/1d462bb2652eccdc160cc13d4218eafbfba035e7

SanderDevisscher commented:
@SaarJunius @soriadelva  I'm not a huge fan of all subcategory icons being the same. It gives heavy "icon not found reverting to default" vibes. However with the 1 click less these should disappear, am I correct ?  Ifso, can't we just keep the old subcategory icons and use the `category-icons.png` as an "icon not found default<sup>1</sup>" until we implement the one click less ?  @soriadelva @RuttenAnneleen @jimcasaer @LynnPallemaerts what do you think ? 

<sup>1</sup>If I'm not mistaken 1 subcategory icon was missing prior to https://github.com/inbo/reporting-rshiny-grofwildjacht/commit/1d462bb2652eccdc160cc13d4218eafbfba035e7

<img width="2517" height="1038" alt="image" src="https://github.com/user-attachments/assets/71e5e6f1-e396-4d83-9679-f1ab6a7feea4" />

SanderDevisscher committed:
Revise barCostUI description in uiText.csv

Updated the description for barCostUI and added more details about cost estimation for wildlife damage cases.

SaarJunius committed:
Revert to custom subcategory card figures

SanderDevisscher commented:
Something strange is going on with the period slider at [Jaarlijks aantal gerapporteerde embryo's voor vrouwelijke wilde zwijnen](https://wbe-uat.inbo.be/app_proxy/8336ef50-0fce-45f5-99e4-548a78d5f901/#) on wbe-uat. When I select "wild zwijn" & wbe "De Zandhaas" & set all sources to INBO I get this:
<img width="434" height="129" alt="image" src="https://github.com/user-attachments/assets/bb340db1-0d51-4fa6-b746-59c309fff832" />

No duplication of years is to be expected

SanderDevisscher commented:
Possibly related is this loop:
![Bezig met opnemen 2025-10-08 135742](https://github.com/user-attachments/assets/75ce5b9e-ac31-407f-9ef1-05870d76da4f)
"damhert" & wbe "De Zandhaas" & set all sources to INBO except geslacht. Switching to any other sources setting stops the loop.

SanderDevisscher commented:
Both issues appear in PRD

SaarJunius commented:
> Possibly related is this loop: "damhert" & wbe "De Zandhaas" & set all sources to INBO except geslacht. Switching to any other sources setting stops the loop.

The loop I can stop, but the issue of duplicated years in the slider is purely a visualisation issue. Because the range is so slow, the intermediate values (between 2022 and 2023) also get a "tick"/vertical line and a value, which rounded becomes either 2022 and 2023 -> Hence the duplicated labels. I cannot solve this myself (except hide the ticks entirely, so let me know if you would want this), but I can raise an issue within the shiny github (although I don't know how quickly this might get fixed).

Example of a slider with no ticks present
<img width="341" height="85" alt="image" src="https://github.com/user-attachments/assets/f19dba95-1636-4227-b66f-27fb1ff450fe" />

FYI: I think there actually exists a similar issue already https://github.com/rstudio/shiny/issues/2472, open since 2019 but no answers yet

SaarJunius committed:
Remove infinite loop

SanderDevisscher commented:
Ok since the issue is present in PRD we can leave it for now. But keep an eye on shiny development into the future.

SanderDevisscher commented:
All issues seem to be fixed but I'm waiting for the switch of the reewild module expected tomorrow to approve and merge

SanderDevisscher commented:
@SaarJunius can you add a period slider to [Jaarlijkse kosten voor schadegevallen](https://faunabeheer-uat.inbo.be/app_proxy/e8409588-9385-4d95-8d69-5ebaa71435e9/?specie=Wild%20zwijn&category=schade&subcategory=schade-kosten&plot=barCostUI&gbifkey=7705930#) ?

SanderDevisscher commented:
@SaarJunius in [Evolutie gerapporteerd afschot per leeftijdscategorie](https://wbe-uat.inbo.be/app_proxy/00365779-8e29-4596-a099-80b79a7097d8/#) on the private section the leeftijdscategorie "Frisling (<6m)" & "Frisling (>6m)" are missing when you switch to source "INBO". Seems not to be a issue on the public section nor for other graphs on the private page with data source INBO for leeftijdscategorie. 

IS (wild zwijn wbe De Zandhaas):
<img width="1300" height="800" alt="newplot (8)" src="https://github.com/user-attachments/assets/c502a181-52c9-4ada-b50c-66539481b386" />

SHOULD BE (wild zwijn Vlaanderen):
<img width="1300" height="800" alt="newplot (9)" src="https://github.com/user-attachments/assets/a5e82656-09b9-4535-b179-599b10740f61" />

SaarJunius committed:
Add period slider to barCost

SaarJunius committed:
Resolve issue with missing zwijn categories on wbe

SanderDevisscher reviewed (approved):
Good work @SaarJunius
